internal/cmd/controller/gitops/reconciler/gitrepo_status.go

```diff
@@ -1,33 +1,53 @@
-package grutil
+package reconciler
 
 import (
 	"context"
 	"fmt"
 	"sort"
 	"strings"
-	"time"
 
 	"github.com/rancher/fleet/internal/cmd/controller/summary"
-	"github.com/rancher/fleet/internal/metrics"
 	fleet "github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+
 	"github.com/rancher/wrangler/v3/pkg/condition"
 	"github.com/rancher/wrangler/v3/pkg/kstatus"
+
 	batchv1 "k8s.io/api/batch/v1"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/equality"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/client-go/util/retry"
-
-	fleetutil "github.com/rancher/fleet/internal/cmd/controller/errorutil"
-	errutil "k8s.io/apimachinery/pkg/util/errors"
 	"sigs.k8s.io/cli-utils/pkg/kstatus/status"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func SetStatusFromBundles(ctx context.Context, c client.Client, gitrepo *fleet.GitRepo) error {
+func setStatus(ctx context.Context, c client.Client, gitrepo *fleet.GitRepo) error {
+	err := setStatusFromBundleDeployments(ctx, c, gitrepo)
+	if err != nil {
+		return err
+	}
+
+	err = setStatusFromBundles(ctx, c, gitrepo)
+	if err != nil {
+		return err
+	}
+
+	if gitrepo.Status.GitJobStatus != "Current" {
+		gitrepo.Status.Display.State = "GitUpdating"
+	}
+
+	setResourceKey(ctx, c, gitrepo)
+
+	gitrepo.Status.Display.ReadyBundleDeployments = fmt.Sprintf("%d/%d",
+		gitrepo.Status.Summary.Ready,
+		gitrepo.Status.Summary.DesiredReady)
+
+	setCondition(&gitrepo.Status, nil)
+
```

Comment:
this is also done in line 224, right after calling this function. I guess it remained when copying/pasting...

Reply:
Ah, right. The "setStatus" code doesn't even change the GitRepoAcceptCondition.